### DISCUSSION TITLE
Ace commands taking too long to finish when inside a Docker container

### DISCUSSION THREAD

**Discussion by iagobruno** [adonisjs/core#2903] [Help] [answered]
Hi everyone,
I'm trying to create a Adonis server using Docker and ace commands are taking TOO LONG to finish (over 8 minutes each) but it doesn't happen when I run it on my machine outside container. 🤔

This project I'm creating uses the latest version of all Adonis packages.
I recently installed Docker on a server that uses a few months ago version of Adonis 5 (when we had to run `node ace build` to compile typescript files first other commands) and I haven't had this experience.

Is anyone else noticing this?

**Comment by thetutlage** [comment]
How are you running the ace commands? Do you run the container in interactive mode and then run the commands?

**Reply by iagobruno** [reply]
I created an entrypoint.sh file in the root of the project with `node ace serve --watch` command.
In interactive mode the commands also take a long time.

[Here's a demo](https://github.com/iagobruno/ace-docker-poor-perf) for you try to replicate on your machine.
And [here's another project](https://github.com/iagobruno/brincando-com-docker) with the same setup but using the old build command.

**Comment by RomainLanz** [comment]
Hey @iagobruno! 👋 

What OS are you using?
Depending on how you are running your container (and sharing files with the OS), it may be really slow.

**Reply by iagobruno** [reply]
I was using Docker Desktop on Windows to run the container.

I recently found that [file sharing between Windows and Linux is quite slow and should be avoided](https://docs.docker.com/docker-for-windows/wsl/#best-practices). This is what was causing the poor ace command experience.

I'm new in Docker world. 😅

**Comment by iagobruno** [comment, marked as answer]
## The answer

I was using Docker Desktop on Windows and recently found that [file sharing between Windows and Linux is known to be slow should be avoided](https://docs.docker.com/docker-for-windows/wsl/#best-practices).
Ace caches the compiled version of .ts files inside node_modules folder and this was causing the poor experience.

Now I'm running Docker directly in WSL in a folder on Linux system and not having this problem anymore.
